tiles/src/main/java/com/google/android/horologist/tiles/images.kt

```
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.horologist.tiles
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.drawable.BitmapDrawable
+import androidx.wear.tiles.ResourceBuilders
+import androidx.wear.tiles.ResourceBuilders.ImageResource
+import coil.ImageLoader
+import coil.request.ImageRequest
+import java.nio.ByteBuffer
+
+/**
+ * Load a Bitmap from a CoilImage loader.
+ *
+ * @param context the context of the service or activity.
+ * @param data the image to fetch in one of the support Coil formats such as String, HttpUrl.
+ * @param configurer any additional configuration of the ImageRequest being built.
+ */
+public suspend fun ImageLoader.loadImage(
+    context: Context,
+    data: Any?,
+    configurer: ImageRequest.Builder.() -> Unit = {}
+): Bitmap? {
+    val request = ImageRequest.Builder(context)
+        .data(data)
+        .apply(configurer)
+        .allowRgb565(true)
+        .allowHardware(false)
+        .build()
+    val response = execute(request)
+    return (response.drawable as? BitmapDrawable)?.bitmap
```

I think this needs better error handling. If the response is `SuccessResult`, the drawable is non-null, but if it's an `ErrorResult`, this method will just swallow the error and return null. I'd argue an exception is a better idea in the latter case, especially since this is a couroutine.

---

Ooooffff, this is tough.  

I'm not certain about this.  ImageRequest has placeholder and error images.  If you set an error image you should get that rather than null.

It's also more inline with Coil compose usage.  I think null is the right thing here.  Even logging this isn't very useful, networks a patchy, you probably have to assume requests fail routinely and decide how to handle it.  Logging isn't something you would actually want generally.

I'll add a test for this.

---

Yes, but the result is generally the Drawable.

```
sealed class ImageResult {
    abstract val drawable: Drawable?
    abstract val request: ImageRequest
}
```

It can be null if it fails and you don't set an error image.

---

I don't want an exception on ErrorImage, I want the drawable.